src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -519,21 +514,41 @@ public byte[] Http2AltSvcOriginUri
             Socket? socket = null;
             SslStream? sslStream = null;
             TransportContext? transportContext = null;
+            Http2Connection[]? lastHttp2Connections = _http2Connections;
 
             // Serialize creation attempt
             await _http2ConnectionCreateLock.WaitAsync(cancellationToken).ConfigureAwait(false);
             try
             {
-                if (_http2Connection != null)
+                Http2Connection[]? currentHttp2Connections = _http2Connections;
+                if (EnableMultipleHttp2Connections)
```

Comment:
It seems like we could just call `GetHttp2ConnectionAcceptingNewStreams` here.

Reply:
Yes, it can be done, but I believe we should not check connection expiration in case `GetHttp2ConnectionAcceptingNewStreams` is called from here because
1. If the connection count is still under maximum and all existing connections are busy, it will be faster to create a new one
2. Checking expiration doesn't seem as a good thing to do while holding a lock here

Reply:
I don't understand your point 1, can you explain?

Re 2, you mean the connection create lock? I'm not worried about this; any cost here will be dwarfed by the time we spend creating the connection.

I think it's fine to recheck the expiration here; it doesn't do any harm, and arguably is more correct, since we'll ensure we don't use an expired connection.

But my main concern here is just to reduce code duplication. Let's have one piece of logic (i.e. GetHttp2ConnectionAcceptingNewStreams) that scans the connection list and returns a connection to use. If you feel strongly about not doing the expiration here, then you can refactor the code to do the expiration check outside of GetHttp2ConnectionAcceptingNewStreams.

Reply:
That said, the more I think about this, the more I think we should do the expiration check in this case too. There's not really any downside that I see, and it's more correct and consistent.

Reply:
BTW, minor point: Calling this function `GetHttp2ConnectionAcceptingNewStreams` is a little confusing because in the single-connection case, the connection may not actually be accepting new streams. 

Suggestion: Call it `GetExistingHttp2Connection` or something like that.

Reply:
Ah, sorry 1. is unrelated here, please disregard.
I renamed the method to `GetExistingHttp2Connection` and got rid of the duplicate code here. Expiration is now always checked.